### DISCUSSION TITLE
Preloading app css

### DISCUSSION THREAD

**Discussion by h3** [quasarframework/quasar#9544] [General - Components / Directives / etc]
Hello, I'm working with Quasar 2 which is able to preload JavaScript fine, but how do I preload the main app css?

In other words, going from:

```html
<link href="/css/app.67170ea4.css" rel="stylesheet">
```
To:

```html
<link rel="preload" href="/css/app.67170ea4.css" as="style">
```

Thanks

**Comment by ontwikkelfabriek** [comment]
@h3 

You could take a look at this:
https://next.quasar.dev/quasar-cli/handling-webpack#postcss

**Reply by h3** [reply]
I already use PostCSS with PurgeCSS, which forced me to use the CssMinimizerPlugin webpack plugin because the CSS was not minified after processing anymore. That said I haven't seen any options regarding preloading..

The correct way to do it would likely to use [Webpack magic comments](https://medium.com/webpack/link-rel-prefetch-preload-in-webpack-51a52358f84c), but that only work with the `import()` syntax. Not sure how it can be applied to the main app chunk.

**Reply by ontwikkelfabriek** [reply]
@h3 

(I am no webpack expert)  But this Quasar app does manage to `preload`  the `app.css` link. Using the `preload-webpack-plugin` inside `chainWebpack` in `quasar.conf`.

The repo:
https://github.com/stelace/marketplace-demo

The site with `preload` css link ( see view sources ):
https://marketplace.demo.stelace.com/

btw that site scores almost a `perfect 100` in `all` Chrome's Lighthouse categories. I have never seen that before in any Quasar app/site.

**Reply by h3** [reply]
Pretty neat project, thanks!
